include/openenclave/edger8r/common.h

```diff
@@ -80,16 +80,16 @@ OE_INLINE oe_result_t oe_add_size(size_t* total, size_t size)
  * Compute and set the pointer value for the given parameter within the input
  * buffer. Make sure that the buffer has enough space.
  */
-#define OE_SET_IN_POINTER(argname, argsize)                                  \
-    if (pargs_in->argname)                                                   \
-    {                                                                        \
-        *(uint8_t**)&pargs_in->argname = input_buffer + input_buffer_offset; \
-        OE_ADD_SIZE(input_buffer_offset, (size_t)(argsize));                 \
-        if (input_buffer_offset > input_buffer_size)                         \
-        {                                                                    \
-            _result = OE_BUFFER_TOO_SMALL;                                   \
-            goto done;                                                       \
-        }                                                                    \
+#define OE_SET_IN_POINTER(argname, argsize, argtype)                       \
+    if (pargs_in->argname)                                                 \
+    {                                                                      \
+        pargs_in->argname = (argtype)(input_buffer + input_buffer_offset); \
```

Comment:
@HernanGatta I remember that you ran into an issue with GCC code generation for ARM and had to fix it in your copy of these macros. With this PR we are going to pass in the type to the macros so that the cast is correctly typed.
Hopefully this should fix the issue that you ran into as well.